Install morpheus-dfp conda package in the DFP container

AnuradhaKaruppiah opened the pull request:
Earlier the morpheus-pipeline stage was using the morpheus-runtime container as the base image. This change switches the base container to nvidia/cuda and installs the morpheus-dfp library.

There is a minor docs fixup in conda_packages.md that I pull in based on the review comments.

AnuradhaKaruppiah commented:
/ok to test

AnuradhaKaruppiah commented:
/ok to test

dagardner-nv commented on examples/digital_fingerprinting/production/Dockerfile:
```diff
@@ -13,35 +13,83 @@
-# Set the workdir to the DFP base folder
-WORKDIR /workspace/examples/digital_fingerprinting/
+# Default MORPHEUS_DFP_VERSION
+ARG MORPHEUS_DFP_VERSION=24.10.00a
```
This should default to the release version, with the possibility of being overridden

```suggestion
ARG MORPHEUS_DFP_VERSION=24.10.00
```

dagardner-nv commented on examples/digital_fingerprinting/production/Dockerfile:
```diff
@@ -13,35 +13,83 @@
-RUN source activate morpheus \
-    && /opt/conda/bin/conda env update --solver=libmamba -n morpheus --file /workspace/conda/environments/examples_cuda-125_arch-x86_64.yaml
+RUN source activate morpheus-dfp \
+     && mamba install -y -c conda-forge tini=0.19
```
Merge this in with the install on line 61-63 to limit the number of times we run conda

AnuradhaKaruppiah replied:
Ack

dagardner-nv commented on examples/digital_fingerprinting/production/Dockerfile:
```diff
@@ -13,35 +13,83 @@
+RUN echo "MORPHEUS_DFP_VERSION is set to: ${MORPHEUS_DFP_VERSION}"
+ENV MORPHEUS_DFP_VERSION=${MORPHEUS_DFP_VERSION}
+RUN source activate morpheus-dfp \
+    && mamba install -y \
```
We've been running into issues with `mamba` lately, and have settled on using `conda --solver=libmamba` as a replacement.

AnuradhaKaruppiah replied:
Ack

dagardner-nv commented on examples/digital_fingerprinting/production/Dockerfile:
```diff
@@ -13,35 +13,83 @@
+
+# Install additional dependencies needed to run the example.
+# Note: this call is very slow and is the longest part of the build proocess - fixme
+RUN source activate morpheus-dfp && /opt/conda/bin/conda env update --solver=libmamba -n morpheus-dfp --file ./conda/dfp_example_cuda-125_arch-x86_64.yaml
```
Any reason why tini, morpheus-dfp, and the requirements aren't all rolled into the yaml? That would allow us to create the env with a single command.

dagardner-nv commented on examples/digital_fingerprinting/production/README.md:
````diff
@@ -27,19 +27,17 @@ Key Features:
 ```
+You can optionally specify the version of the morpheus-dfp conda package for the pipeline via the build-arg. The version defaults to the latest release.
+```bash
+docker compose build --build-arg MORPHEUS_DFP_VERSION=24.10.00a
````
We should probably go back to using `git describe` to drive this, otherwise it will need to be added to `ci/release/update-version.sh`

AnuradhaKaruppiah replied:
Makes sense. But in that case I should probably leave morpheus-dfp installation in this docker file.

AnuradhaKaruppiah commented:
@dagardner-nv I have addressed your comments. I also want to highlight the boto3 dependency change to set the version.

If I run with the boto3 version set the solver takes  half the time -
```
=> [morpheus_pipeline base 7/7] RUN source activate morpheus-dfp && /opt/conda/bin/conda env update --solve  293.1s
```
And this is boto3 without a specific version -
```
=> [morpheus_pipeline base 7/7] RUN source activate morpheus-dfp && /opt/conda/bin/conda env update --solve  637.3s
```

AnuradhaKaruppiah commented:
/ok to test

AnuradhaKaruppiah commented:
/ok to test

dagardner-nv commented on examples/digital_fingerprinting/production/Dockerfile:
```diff
@@ -13,35 +13,50 @@
+COPY . /workspace/examples/digital_fingerprinting/production/
+
+# Install morpheus-dfp and any additional dependencies needed to run the examples
+RUN source activate morpheus-dfp && /opt/conda/bin/conda env update --solver=libmamba -n morpheus-dfp --file ./conda/environments/dfp_example_cuda-125_arch-x86_64.yaml
```
I would merge this in with the `conda create` command on line 37 it should be possible to do:
```bash
conda env create --solver=libmamba -y --name morpheus-dfp --file ./conda/environments/dfp_example_cuda-125_arch-x86_64.yaml
```

AnuradhaKaruppiah replied:
this feels so obvious, now!

AnuradhaKaruppiah commented:
/ok to test

dagardner-nv reviewed (approved):
LGTM, but lets double-check with @mdemoret-nv when we should be tracking all three version digits or just major/minor

AnuradhaKaruppiah commented:
/ok to test
